pylsp_mypy/plugin.py

```diff
@@ -90,25 +90,24 @@ def parse_line(line: str, document: Optional[Document] = None) -> Optional[Dict[
 
         lineno = int(linenoStr or 1) - 1  # 0-based line number
         offset = int(offsetStr or 1) - 1  # 0-based offset
+        end_lineno = (int(endlinenoStr) - 1) if endlinenoStr else None
+        end_offset = (int(endoffsetStr)) if endoffsetStr else None
         errno = 2
         if severity == "error":
             errno = 1
         diag: Dict[str, Any] = {
             "source": "mypy",
             "range": {
                 "start": {"line": lineno, "character": offset},
-                # There may be a better solution, but mypy does not provide end
-                "end": {"line": lineno, "character": offset + 1},
+                "end": {"line": end_lineno or lineno, "character": end_offset},
```

Comment:
Those can be `None` and `None` is not allowed in `Position`.
I see that you later adjust `character` if it's `None` but not `line`.
You should probably fall back to previous behavior in that case.

Reply:
So mypy with this option behaves differently in 3.7? (Tests seem to indicate that, I just want to confirm)
Is this documented in mypy as intended or is this a bug?

Either way we will not write a lengthy workaround as 3.7 support will likely be dropped soon anyway.

Reply:
Didn't find anything in docs or an issue but I did try with multiple different error types and they all had the same behaviour.

Can you clarify what you mean by not supporting a lengthy workaround.

Reply:
Maybe we should ask the mypy devs if this is intentional or a bug.

Mypy seems to behave differently in python 3.7 than in the other versions. In order to handle this, 'unreadable code' (as you put it) is required. This is what I mean by 'lengthy workaround'. Having this in the codebase is not desirable in the first place, but in this case there is another point to consider. Python 3.7 will be end of life very soon. Mypy will most likely drop support for it then and so will we. This means, that in approx two months our code does not need to work with python 3.7 anymore. So there is little reason to try really hard to handle some special case in 3.7 now.

So: If the test for 3.7 does not work, just ignore it for the time being.

Cleaning up the regex is fine with me and probably a good idea.